server/sonar-server/src/main/java/org/sonar/server/qualityprofile/RuleActivator.java

```diff
@@ -290,116 +272,74 @@ private ActiveRuleDto doInsert(ActiveRuleChange change, RuleActivatorContext con
     dao.insert(dbSession, activeRule);
     for (Map.Entry<String, String> param : change.getParameters().entrySet()) {
       if (param.getValue() != null) {
-        ActiveRuleParamDto paramDto = ActiveRuleParamDto.createFor(context.ruleParamsByKeys().get(param.getKey()));
+        ActiveRuleParamDto paramDto = ActiveRuleParamDto.createFor(rule.getParam(param.getKey()));
```

Comment:
![Bug](https://sonarsource.github.io/sonar-github/bug.png 'Bug') Bug: "createFor" is marked "javax.annotation.ParametersAreNonnullByDefault" but parameter 1 could be null. ([squid:S2637](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2637))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.sonarqube%3Asonarqube&branch=feature%2Fsb%2FSONAR-10052%2Fqprofiles&issues=AWE-qvP5mckxG3vIth74&open=AWE-qvP5mckxG3vIth74)

[ISSUE_KEY]: AWE-qvP5mckxG3vIth74